pkg/prometheus/agent/statefulset_test.go

```diff
@@ -206,3 +208,87 @@ func makeStatefulSetFromPrometheus(p monitoringv1alpha1.PrometheusAgent) (*appsv
 		0,
 		nil)
 }
+
+func TestPodTopologySpreadConstraintWithAdditionalLabels(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		spec monitoringv1alpha1.PrometheusAgentSpec
```

Comment:
If possible, I would define the expected topology spread constraints in the test struct.
```suggestion
		spec monitoringv1alpha1.PrometheusAgentSpec
		expected []v1.TopologySpreadConstraint
```